packages/cache/src/internal/tar.ts

```diff
@@ -14,14 +14,20 @@ async function getTarPath(
   switch (process.platform) {
     case 'win32': {
       const systemTar = `${process.env['windir']}\\System32\\tar.exe`
-      if (compressionMethod !== CompressionMethod.Gzip) {
+      const gnuTar = `${process.env['PROGRAMFILES']}\\Git\\usr\\bin\\tar.exe`
+      if (existsSync(gnuTar)) {
+        // Use GNUtar as default on windows
+        args.push('--force-local')
+        return gnuTar
+      } else if (
+        compressionMethod !== CompressionMethod.Gzip ||
```

Comment:
This line might short circuit to add `--force-local` for `bsdtar`. The `else if` condition is reached when `existsSync(gnuTar)` is false. Now if the compression method is not `gzip`, `await utils.isGnuTarInstalled()` might not be evaluated.

Reply:
Thats why I have still not removed the condition check if only bsdtar exists then use gzip. Can still remove that condition as per our discussion as this is not a release per se but I would still want to keep things not breaking.

Reply:
talking about this condition: https://github.com/actions/toolkit/blob/main/packages/cache/src/internal/cacheUtils.ts#L93

Reply:
That makes sense. But we should still fix the logic here so it is not possible to have `--force-local` added incase of `bsdtar`. Since this might happen as soon as we remove the aforementioned condition in subsequent PRs.

Reply:
Yes. That should be handled when we remove the condition then.

Reply:
Sounds good. The subsequent PR should remove all `gzip` related logic.

Reply:
>But we should still fix the logic here so it is not possible to have --force-local added incase of bsdtar

I came here to add same comment. This is adding `'--force-local'` for bsdtar. I would suggest to keep appropriate behaviour even though you want to change it in next PR.

Reply:
Currently incase of bsdtar this condition won't get hit as bsdtar only does compression with gzip